src/shared/mount-util.h

```diff
@@ -118,18 +118,23 @@ int mount_image_in_namespace(pid_t target, const char *propagate_path, const cha
 
 int make_mount_point(const char *path);
 
-typedef enum RemountIdmapFlags {
+typedef enum RemountIdmapping {
+        REMOUNT_IDMAPPING_NONE,
         /* Include a mapping from UID_MAPPED_ROOT (i.e. UID 2^31-2) on the backing fs to UID 0 on the
          * uidmapped fs. This is useful to ensure that the host root user can safely add inodes to the
          * uidmapped fs (which otherwise wouldn't work as the host root user is not defined on the uidmapped
          * mount and any attempts to create inodes will then be refused with EOVERFLOW). The idea is that
          * these inodes are quickly re-chown()ed to more suitable UIDs/GIDs. Any code that intends to be able
          * to add inodes to file systems mapped this way should set this flag, but given it comes with
          * certain security implications defaults to off, and requires explicit opt-in. */
-        REMOUNT_IDMAP_HOST_ROOT = 1 << 0,
-} RemountIdmapFlags;
-
-int remount_idmap(const char *p, uid_t uid_shift, uid_t uid_range, RemountIdmapFlags flags);
+        REMOUNT_IDMAPPING_HOST_ROOT,
+        /* Include a mapping from root user within the container to the owner of the bind mounted directory.
+         * This ensure no root-owned files will be written in a bind-mounted directory owned by a different
+         * user. */
+        REMOUNT_IDMAPPING_HOST_OWNER,
```

Comment:
For all our non-flag enums, we add two extra entries to the end:

```c
_REMOUNT_IDMAPPING_MAX,
_REMOUNT_IDMAPPING_INVALID = -EINVAL,
```

Simply as a matter of coding style. Please do so here too. It's useful once you intend to pass an invalid value in the enum, or want to know the valid rnage.

Reply:
not actually addressed?